### ISSUE TITLE
Backslash escaped incorrectly when entering visual mode

### ISSUE THREAD

**Issue by anirban-mukherjee** [rstudio/rstudio#10585]
### System details

    RStudio Edition : Desktop
    RStudio Version :  2022.02.0 Build 423
    OS Version      :  Mac OS 12.1
    R Version       : 4.1.2

### Steps to reproduce the problem

In a new R Notebook add the following lines of text.

```
Refer to \S \ref{a}.
Refer to \S \\ref{a}.
Refer to \ref{a}.
```

Switch from source mode to visual model and then back to source mode.

### Describe the problem in detail

When entering visual model, RStudio escapes the backslash in the first sentence to make it the same as the second. The first sentence should be legal markdown and remain unchanged. The third sentence, which is also legal markdown, is not affected and remains unchanged.

I have put all three sentences in the MWE to illustrate the issue; this occurs whenever the first sentence is in the document irrespective of the second or third sentence.

### Describe the behavior you expected

The sentences should remain unchanged.

- [X ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [X ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [X ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [X ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by anirban-mukherjee** [comment]
This seems to happen whenever two accents/symbols are used consecutively. E.g.: `\S \pounds`. RStudio escapes the second accent/symbol when going from source to visual to source mode.

Also I find that whenever accents are used in the text (e.g., `\"{o}`), entering and returning from visual mode removes the backslash. I don't know if this issue is related to the earlier issue.

**Comment by jjallaire** [comment]
Should be fixed here: https://github.com/rstudio/rstudio/commit/3bd40274936a1a0353ad75f824e020945823865c

**Comment by romainfrancois** [comment]
on dev ()

With this Rmd: 

````md
---
title: "R Notebook"
output: html_notebook
---

Refer to \S \ref{a}.
Refer to \S \\ref{a}.
Refer to \ref{a}.
````

<img width="427" alt="image" src="https://user-images.githubusercontent.com/2625526/167136399-ca07546a-678b-4083-a308-b7f4cd826450.png">

switching to visual does: 

<img width="603" alt="image" src="https://user-images.githubusercontent.com/2625526/167136801-a5543686-3fdb-4c64-909e-524eb7a7c4f2.png">

and then back to source; 

<img width="548" alt="image" src="https://user-images.githubusercontent.com/2625526/167136838-6753ce22-b7e3-4caf-833f-77700df73cb3.png">

**Comment by romainfrancois** [comment]
So the original issue does seem fixed, but lines get wrapped.

**Comment by jjallaire** [comment]
The change in line wrapping is also something pandoc introduces (because there are no soft line breaks in the original source).

FWIW we don't currently promise that we will round-trip source (as it has to go through the pandoc AST and back out). Here are some of the known transformations that occur: https://rstudio.github.io/visual-markdown-editing/markdown.html#markdown-output

**Comment by romainfrancois** [comment]
Thanks for the clarification. In that case I can confirm this is fixed with: RStudio 2022.06.0-daily+369. 

<img width="354" alt="image" src="https://user-images.githubusercontent.com/2625526/167243479-12b4161b-1fb0-419c-8b58-5a83cf5e4b1d.png">

<img width="551" alt="image" src="https://user-images.githubusercontent.com/2625526/167243489-b93681ab-b983-4b8b-bd1a-a5b0cc29c3b1.png">
